avid-adventurers/src/pages/Home.tsx

```diff
@@ -1,19 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import { EventBox } from '../components/EventBox/EventBox';
 import { useOnboarding } from '../utils/onboardingContext';
 import { NavBar } from '../components/NavBar/NavBar';
-import { Container, FormWrapperLeft, ScrollContainer, Title } from './styles';
+import { Container, FormWrapperLeft, ScrollContainerEvents, ScrollContainerPeople, Title, Title3, SearchRow, SearchInputContainer, SearchIconImage, SearchInputField } from './styles';
 import { ProfileBox } from '../components/ProfileBox/ProfileBox';
 import { eventData } from '../utils/eventData';
+import { userData } from '../utils/userData';
+import searchIcon from '../assets/search.svg';
+
 
 const Events: React.FC = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const filteredEvents = Object.entries(eventData).filter(([_, event]) =>
+    event.title.toLowerCase().includes(searchTerm.toLowerCase())
+  );
+
   return (
     <Container>
       <FormWrapperLeft>
-        <Title>Top Events</Title>
-        <ScrollContainer>
-          {Object.entries(eventData).map(([key, event]) => (
+        <SearchRow>
+          <Title3>Top Events</Title3>
```

Comment:
Instead of a SearchRow, let's refer to and utilize the Figma design (just add a filter icon + search icon aligned side by side in a row). 
<img width="291" alt="Screenshot 2025-05-06 at 3 15 04 PM" src="https://github.com/user-attachments/assets/de130831-f75c-4674-b8b5-27aa950f6a43" />